compiler.js

```diff
@@ -3,6 +3,62 @@ function isWholeProgramTarget(compileTarget)
     return compileTarget == "METAL" || compileTarget == "SPIRV";
 }
 
+const imageMainSource = `
+import user;
+
+uniform float time;
+RWStructuredBuffer<int>               outputBuffer;
+[format("r32f")] RWTexture2D<float>   texture;
+
+inline float encodeColor(float4 color)
+{
+    uint4 colorInt = { uint(color.x * 255.0f),
+                       uint(color.y * 255.0f),
+                       uint(color.z * 255.0f),
+                       uint(color.w * 255.0f) };
+
+    float encodedColor = float(colorInt.x << 24 | colorInt.y << 16 | colorInt.z << 8 | colorInt.w);
+    return encodedColor;
+}
+
+[shader("compute")]
+void imageMain(uint3 dispatchThreadID : SV_DispatchThreadID)
+{
+    uint width = 0;
+    uint height = 0;
+    texture.GetDimensions(width, height);
+    float4 color = imageMain(dispatchThreadID.xy, int2(width, height), time);
```

Comment:
improve: instead of complicating the parameter list of `imageMain`, we can make it so that the `time` can be queried by a `getTime()` function defined in `playground` module.

Reply:
yea, follow-up.